### PR TITLE
feat: support customized initialization SQL statements

### LINKED ISSUE #268: How to init sql commands at boot time
Hi, I want to init some sql commands at docker instance startup, such as load extensions and declare secrects, may you help ?

### PR DESCRIPTION
**Pull request by VWagen1989**
* resolve #268 

This PR enables MyDuck Server running in the docker container to execute the SQL statements placed in the files under either '/docker-entrypoint-initdb.d/mysql' or '/docker-entrypoint-initdb.d/postgres'. The SQL files in the former one will be executed based on MySQL protocol, while the latter will be performed under PostgreSQL protocol.

Here's an example illustrates how to install DuckDB extensions and create temporary secret in memory automatically once MyDuck Server is ready.

* Creating MyDuck Server container without a SQL file, the extension 'aws' is not loaded and no secrets can be found.
```bash
vwagen@wuxiangdeMacBook-Pro-2 init_sqls % docker run -d -p 13306:3306 -p 15432:5432 --name=myduck apecloud/myduckserver:latest                                                                                                                                                                                                  
199a368744d314a1db2e498082767e0c2b800c440f913fc59930c1bd4424b125
vwagen@wuxiangdeMacBook-Pro-2 init_sqls % 
vwagen@wuxiangdeMacBook-Pro-2 init_sqls % 
vwagen@wuxiangdeMacBook-Pro-2 init_sqls % docker exec myduck psql -h 127.0.0.1 -U postgres -c "SELECT extension_name, installed, description FROM duckdb_extensions()"
  extension_name  | installed |                                    description                                     
------------------+-----------+------------------------------------------------------------------------------------
 arrow            | t         | A zero-copy data integration between Apache Arrow and DuckDB
 autocomplete     | f         | Adds support for autocomplete in the shell
 aws              | f         | Provides features that depend on the AWS SDK
 azure            | f         | Adds a filesystem abstraction for Azure blob storage to DuckDB
 delta            | f         | Adds support for Delta Lake
 excel            | f         | Adds support for Excel-like format strings
 fts              | f         | Adds support for Full-Text Search Indexes
 httpfs           | f         | Adds support for reading and writing files over a HTTP(S) connection
 iceberg          | f         | Adds support for Apache Iceberg
 icu              | t         | Adds support for time zones and collations using the ICU library
 inet             | f         | Adds support for IP-related data types and functions
 jemalloc         | t         | Overwrites system allocator with JEMalloc
 json             | t         | Adds support for JSON operations
 motherduck       | f         | Enables motherduck integration with the system
 mysql_scanner    | f         | Adds support for connecting to a MySQL database
 parquet          | t         | Adds support for reading and writing parquet files
 postgres_scanner | t         | Adds support for connecting to a Postgres database
 spatial          | f         | Geospatial extension that adds support for working with spatial data and functions
 sqlite_scanner   | f         | Adds support for reading and writing SQLite database files
 substrait        | f         | Adds support for the Substrait integration
 tpcds            | f         | Adds TPC-DS data generation and query support
 tpch             | f         | Adds TPC-H data generation and query support
 vss              | f         | Adds indexing support to accelerate Vector Similarity Search
(23 rows)

vwagen@wuxiangdeMacBook-Pro-2 init_sqls % docker exec myduck psql -h 127.0.0.1 -U postgres -c "SELECT name, type, provider, persistent, storage, secret_string FROM duckdb_secrets()"
 name | type | provider | persistent | storage | secret_string 
------+------+----------+------------+---------+---------------
(0 rows)
```

* If we place some SQL statements about installing extension and creating secret for DuckDB in the file 'init.sql', and then mount the file to '/docker-entrypoint-initdb.d/postgres/init.sql' to the MyDuck Server container, we will find that extension will be installed and a secret will be created.
```bash
vwagen@wuxiangdeMacBook-Pro-2 init_sqls % cat init.sql                                                                                                                                                                                                                                                                      
INSTALL aws;                                                                                                                                                                                                                                                                                                                
CREATE SECRET my_secret (                                                                                                                                                                                                                                                                                                   
    TYPE S3,                                                                                                                                                                                                                                                                                                                
    KEY_ID 'my_secret_key',                                                                                                                                                                                                                                                                                                 
    SECRET 'my_secret_value',                                                                                                                                                                                                                                                                                               
    REGION 'my_region'                                                                                                                                                                                                                                                                                                      
);                                                                                                                                                                                                                                                                                                                          
vwagen@wuxiangdeMacBook-Pro-2 init_sqls % docker run -d -p 13306:3306 -p 15432:5432 --name=myduck -v ./init.sql:/docker-entrypoint-initdb.d/postgres/init.sql apecloud/myduckserver:latest                                                                                                                                       
383e6c1b66e1a46b2f2516cd7cb77a5185da62291f4656d9a86aef16802c37b5                                                                                                                                                                                                                                                            
vwagen@wuxiangdeMacBook-Pro-2 init_sqls %                                                                                                                                                                                                                                                                                   
vwagen@wuxiangdeMacBook-Pro-2 init_sqls % docker exec myduck psql -h 127.0.0.1 -U postgres -c "SELECT extension_name, installed, description FROM duckdb_extensions()"                                                                                                                                                      
  extension_name  | installed |                                    description                                                                                                                                                                                                                                              
------------------+-----------+------------------------------------------------------------------------------------                                                                                                                                                                                                         
 arrow            | t         | A zero-copy data integration between Apache Arrow and DuckDB
 autocomplete     | f         | Adds support for autocomplete in the shell
 aws              | t         | Provides features that depend on the AWS SDK
 azure            | f         | Adds a filesystem abstraction for Azure blob storage to DuckDB
 delta            | f         | Adds support for Delta Lake
 excel            | f         | Adds support for Excel-like format strings
 fts              | f         | Adds support for Full-Text Search Indexes
 httpfs           | t         | Adds support for reading and writing files over a HTTP(S) connection
 iceberg          | f         | Adds support for Apache Iceberg
 icu              | t         | Adds support for time zones and collations using the ICU library
 inet             | f         | Adds support for IP-related data types and functions
 jemalloc         | t         | Overwrites system allocator with JEMalloc
 json             | t         | Adds support for JSON operations
 motherduck       | f         | Enables motherduck integration with the system
 mysql_scanner    | f         | Adds support for connecting to a MySQL database
 parquet          | t         | Adds support for reading and writing parquet files
 postgres_scanner | t         | Adds support for connecting to a Postgres database
 spatial          | f         | Geospatial extension that adds support for working with spatial data and functions
 sqlite_scanner   | f         | Adds support for reading and writing SQLite database files
 substrait        | f         | Adds support for the Substrait integration
 tpcds            | f         | Adds TPC-DS data generation and query support
 tpch             | f         | Adds TPC-H data generation and query support
 vss              | f         | Adds indexing support to accelerate Vector Similarity Search
(23 rows)

vwagen@wuxiangdeMacBook-Pro-2 init_sqls % docker exec myduck psql -h 127.0.0.1 -U postgres -c "SELECT name, type, provider, persistent, storage, secret_string FROM duckdb_secrets()"
   name    | type | provider | persistent | storage |                                                              secret_string                                                                
-----------+------+----------+------------+---------+------------------------------------------------------------------------------------------------------------------------------------------
 my_secret | s3   | config   | f          | memory  | name=my_secret;type=s3;provider=config;serializable=true;scope=s3://,s3n://,s3a://;key_id=my_secret_key;region=my_region;secret=redacted
(1 row)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,6 +97,17 @@ This setup exposes:
 - **Port 13306** for MySQL wire protocol connections.
 - **Port 15432** for PostgreSQL wire protocol connections, allowing direct DuckDB SQL.
 
+To initialize MyDuck Server by executing custom SQL statements once it’s ready, you can place the SQL statements in a file with the .sql extension and mount the file to either `/docker-entrypoint-initdb.d/mysql/` or `/docker-entrypoint-initdb.d/postgres/` in the Docker container. The choice between these directories depends on the protocol used for executing the SQL statements.
+
+For instance:
+```bash
+# Execute init.sql on the port 3306 based on MySQL protocol
```
Comment (TianyuZhang1214) [inline README.md:104]: Execute init.sql on MySQL protocol.

File: README.md [hunk 2]
```diff
@@ -97,6 +97,17 @@ This setup exposes:
...
+docker run -d -p 13306:3306 -p 15432:5432 --name=myduck -v ./init.sql:/docker-entrypoint-initdb.d/postgres/init.sql apecloud/myduckserver:latest
+
+# Execute init.sql on the port 5432 based on PostgreSQL protocol
```
Comment (TianyuZhang1214) [inline README.md:107]: Execute init.sql on Postgres protocol.

**Review by TianyuZhang1214 — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -97,6 +97,17 @@ This setup exposes:
 - **Port 13306** for MySQL wire protocol connections.
 - **Port 15432** for PostgreSQL wire protocol connections, allowing direct DuckDB SQL.
 
+To initialize MyDuck Server by executing custom SQL statements once it’s ready, you can place the SQL statements in a file with the .sql extension and mount the file to either `/docker-entrypoint-initdb.d/mysql/` or `/docker-entrypoint-initdb.d/postgres/` in the Docker container. The choice between these directories depends on the protocol used for executing the SQL statements.
```
Comment (fanyang01) [inline README.md:100]: This paragraph should be added as a separate subsection and be placed after the "Replicating Data" subsection.

File: README.md (hunk 1 above)
Comment (VWagen1989) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (VWagen1989) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (VWagen1989) [reply README.md]: done

**Review by TianyuZhang1214 — APPROVED** [review]
LGTM

File: README.md [hunk 4]
```diff
@@ -19,6 +19,7 @@
   - [Installation](#installation)
   - [Usage](#usage)
   - [Replicating Data](#replicating-data)
+  - [Initialize MyDuck Server with Custom SQLs](#initialize-myduck-server-with-custom-sqls)
```
Comment (fanyang01) [inline README.md:22]: Let's make it shorter: `Using Custom Initialization Scripts`. `MySQL Server` is unneeded.

**Pull request merged**